Akmal / MT5 page redesign

akmal-binary opened the pull request.

zaki-binary commented on src/javascript/app/pages/user/metatrader/metatrader.ui.js:
```diff
@@ -722,11 +725,11 @@ const MetaTraderUI = (() => {
         // otherwise they are adding more server to their current account type
         const saved_mt5_account = Client.get('mt5_account');
-        return MetaTraderConfig.getCleanAccType(saved_mt5_account);
+        return saved_mt5_account;
```
Can simplify with just :
```suggestion
        return Client.get('mt5_account');
```

akmal-binary replied:
Yes, nice catch 👍🏻

zaki-binary commented on src/javascript/app/pages/user/metatrader/metatrader.ui.js:
```diff
@@ -814,7 +817,7 @@ const MetaTraderUI = (() => {
         Object.keys(accounts_info).sort(sortMt5Accounts).forEach(acc_type => {
             // remove server from name
-            const clean_acc_type = MetaTraderConfig.getCleanAccType(acc_type);
+            const clean_acc_type =  MetaTraderConfig.getCleanAccType(acc_type, 2);
```
Additional space?

matin-binary commented on src/sass/app/metatrader.scss:
```diff
@@ -211,13 +211,13 @@
+        .add_region {
             position: absolute;
-            bottom: 8px;
-            right: 8px;
+            bottom: 16px;
+            right: 21px;
 
             span {
-                padding: 6px 8px;
+                padding: 9px;
```
Nit, can we use `rem` instead of `px`

akmal-binary replied:
@matin-binary I believe on binary static we do not define the constant size of `rem` unit, therefore, we use pixels across all the styles. Let's keep it this way for consistency.

gareth-deriv commented on src/javascript/app/pages/user/metatrader/metatrader.ui.js:
```diff
@@ -63,6 +63,78 @@ const MetaTraderUI = (() => {
+
+                if (is_used_server) {
+                    num_servers.used += 1;
+                    label_text += localize(' (account created)');
```
Nitpick, and possibly against existing convention:
I would have expected
```
' ('+ localize('account created') +')'
```

gareth-deriv reviewed (commented):
Great work 👍  Thanks for the invite to review your work.

I've left only nitpicks.
* One theme is to comment to explain why the code is doing something (not what it's doing),
in a few places, to help new devs (and reviewers) check for mis-understandings in future
* I felt that the `getCleanAccType`, whilst well refactored and abstracted, caters for a more complicated use-case than we actually have, and introduces risk in being so clever. If it's at all possible to simply it, even at the cost of cuteness, I'd go that way.

Your doubt I could help was well founded; Whilst I've added comments, I doubt I'll have caught anything substantial here that would de-risk it for going live.

Sorry I couldn't be of more help, but maybe at least one of the nitpicks might help 😄

akmal-binary commented:
> Great work 👍 Thanks for the invite to review your work.
> 
> I've left only nitpicks.
> 
> * One theme is to comment to explain why the code is doing something (not what it's doing),
>   in a few places, to help new devs (and reviewers) check for mis-understandings in future
> * I felt that the `getCleanAccType`, whilst well refactored and abstracted, caters for a more complicated use-case than we actually have, and introduces risk in being so clever. If it's at all possible to simply it, even at the cost of cuteness, I'd go that way.
> 
> Your doubt I could help was well founded; Whilst I've added comments, I doubt I'll have caught anything substantial here that would de-risk it for going live.
> 
> Sorry I couldn't be of more help, but maybe at least one of the nitpicks might help 😄

Thank you very much for your review Gareth 👍🏻 I've added explanations and applied your suggestions where it was possible.

gareth-deriv commented on src/javascript/app/pages/user/metatrader/metatrader.ui.js:
```diff
@@ -64,10 +64,10 @@ const MetaTraderUI = (() => {
     const populateWebLinks = (server_info) => {
-        const $mt5_web = $('.mt5-web');
-        const $web_links = $mt5_web.find('a');
+        const mt5_url = `https://trade.mql5.com/trade${server_info && `?servers=${server_info.environment}&trade_server=${server_info.environment}`}`;
```
Ah, to clarify my intention here:
```
 const mt5_url = `https://trade.mql5.com/trade`;
```
would go at the top of the file, along with any other immutable constants, so that they're obvious when you look at it from the start.

Just a suggestion / coding-style, though, so feel free to ignore.

akmal-binary replied:
Oh yes, this is a good suggestion. I splitted the domain part of URL from query string and moved it on top of the file. 👍🏻
